### PR TITLE
FRI-123 Update classification PAC/SAC automatically

### PR DESCRIPTION
**Pull request by dmcgihtsdo**
FRI-123 is concerned with updating the PAC/SAC classification criteria on AAG whenever a commit is made. 

The majority of the functionality was written in FRI-61; this PR simply adds support for an edge case where the classification results are empty (as this does not trigger a commit).

### CODE REVIEW

File: src/main/java/org/snomed/snowstorm/core/data/services/BranchMetadataHelper.java [hunk 1]
```diff
@@ -19,6 +19,8 @@
 	// Metadata for internal use only and can't be updated via branch metadata REST service
 	public static final String INTERNAL_METADATA_KEY = "internal";
 
+	public static final String CLASSIFIED_KEY = "classified";
```
Comment (dmcgihtsdo) [inline src/main/java/org/snomed/snowstorm/core/data/services/BranchMetadataHelper.java:22]: I couldn't find a constant for classified, so I added it here beside what I also had to use.

File: src/main/java/org/snomed/snowstorm/core/data/services/classification/ClassificationService.java [hunk 2]
```diff
@@ -242,6 +246,7 @@ private void init() throws ServiceException {
 												final boolean classified = !inferredRelationshipChangesFound && !equivalentConceptsFound;
 												BranchClassificationStatusService.setClassificationStatus(latestBranchCommit, classified);
 												branchService.updateMetadata(latestBranchCommit.getPath(), latestBranchCommit.getMetadata());
+												mockCommitCompletion(latestBranchCommit);
```
Comment (dmcgihtsdo) [inline src/main/java/org/snomed/snowstorm/core/data/services/classification/ClassificationService.java:249]: This method is triggered whenever a classification is completed.

File: src/main/java/org/snomed/snowstorm/core/data/services/classification/ClassificationService.java [hunk 3]
```diff
@@ -438,6 +443,7 @@ public void saveClassificationResultsToBranch(String path, String classification
 				classification.setStatus(SAVED);
 			}
 
+			mockCommitCompletion(path, classification);
```
Comment (dmcgihtsdo) [inline src/main/java/org/snomed/snowstorm/core/data/services/classification/ClassificationService.java:446]: This method is triggered whenever a classification is updated manually.

**Review by kaicode — CHANGES_REQUESTED** [review]
The branch classification status is already being set via `BranchClassificationStatusService.setClassificationStatus`. 
The only case I can think of where a classification status change is not already being sent to the AAG is when classification completes and there are no changes - in this situation the metadata is being updated but we need to post to the AAG. We already do this when creating a branch.

**Comment by dmcgihtsdo** [comment]
> The branch classification status is already being set via `BranchClassificationStatusService.setClassificationStatus`.
> The only case I can think of where a classification status change is not already being sent to the AAG is when classification completes and there are no changes - in this situation the metadata is being updated but we need to post to the AAG. We already do this when creating a branch.

Sorry @kaicode , I don't quite understand. 

I think I'm already doing what you've said? For my testing, I've created a new branch, and then started a classification. The classification finishes with no Relationship changes and ends up at line 248 in ClassificationService. I am then sending a request to AAG with this information.

**Comment by kaicode** [comment]
@dmcgihtsdo I am not suggesting what you have done will not work, I'm just suggesting that we could have more code reuse if you look at the other places where we are doing the same thing. I'll call you.

**Comment by dmcgihtsdo** [comment]
> @dmcgihtsdo I am not suggesting what you have done will not work, I'm just suggesting that we could have more code reuse if you look at the other places where we are doing the same thing. I'll call you.

Thanks for the clarification Kai. 

I have removed the unnecessary call to the mock method (which also contained the redundant constant). Now all that is changed is the call to AAG when an empty classification.

**Review by kaicode — APPROVED** [review]
Looks good to me 👍

**Pull request merged**